### ISSUE TITLE
Feature request: sam generate lambda - Adds a lambda function to existing sam template

### ISSUE THREAD

**Issue by antoineaws** [aws/aws-sam-cli#4498]
### Describe your idea/feature/enhancement

I wish SAM CLI would have a sub-command that generates artifacts and updates the sam template. Similar to the `ionic generate` command (https://ionicframework.com/docs/cli/commands/generate)
e.g. 
- `sam generate lambda FunctionName` - Allows us to add a lambda function to a SAM template and potentially attach it to API GW
- `sam generate api APIGWName ResourceName LambdaFunctionName` - Allows us to a lambda event and attach it to API GW at the right path

### Proposal

As described above, the SAM CLI would be modified to generate simple lambda functions and update the sam template as well.

**Comment by qingchm** [comment]
@antoineaws Thanks for contributing this feature request! We have many similar development ideas and requests so we are interested to understand your experience here. To understand your request better, did you find adding a function too much work for any new additions after init command has finished? If that's the case would you desire a command that adds a function definition into the template yaml and link that to a empty handler function that we create for you? Or did you feel that you don't find any init templates that is close to your project so you want this "generate" behaviour to build from scratch?

**Comment by antoineaws** [comment]
@qingchm 
- "did you find adding a function too much work for any new additions after init command has finished?" correct, currently we have to manually copy paste a function (hopefully we've kept an empty one around, else web search), add it to template.yaml, modify launch.json in the case of nodejs so it can be debugged and remove any unnecessary packages if any extras from the copied lambda.
- "If that's the case would you desire a command that adds a function definition into the template yaml and link that to a empty handler function that we create for you?" - Exactly. 

It's part of the Developer Experience, rather than copying and pasting each time, ideally the generate command would generate an empty handler function, adds to template.yaml and asks the developer whether they want to update launch.json. It could also help with best practices (e.g. showing developers how to have small lambdas with their own pacakge.json rather than 10 different lambdas in the same folder with same package.json and node_modules, leading to a bloated deployment package etc).
Feel free if you need more info. Thanks!

**Comment by moelasmar** [comment]
@antoineaws .. I am not sure if you aware about the new preview service ([Application composer](https://docs.aws.amazon.com/application-composer/latest/dg/getting-started-demo.html)) that you can use to do the same functionality you are asking for but through a visual UI. or do you prefer to have this functionality using CLI tool?

**Comment by antoineaws** [comment]
I would say we need both. From a developer experience(DX) perspective, they would be developing and iterating on their code locally on their machine in the IDE. So having the SAM cli installed locally with that functionality in my opinion is a great DX rather than having to go to the console to visually drag and drop a lambda (which amends the SAM template), and then the developer will have to copy paste the generated code locally to their project. 

Simply put,  `sam generate api APIGWName ResourceName LambdaFunctionName` would be simpler and more streamlined from a DX. 

Now if they need to add services like SQS, SNS, they could use Application Composer to streamline or even also down the line we can add it to `sam generate`..

**Comment by vickywane** [comment]
@moelasmar @qingchm Any update on this feature request?

**Comment by onecompileman** [comment]
Updates on this please, I need this as well to streamline my development
